deepinv/optim/optim_iterators/spectral_methods.py

```diff
@@ -0,0 +1,107 @@
+import torch
+
+from .optim_iterator import OptimIterator, fStep, gStep
+
+
+class SMIteration(OptimIterator):
+    r"""
+    Iterator for Spectral Methods.
+    
+    Class for a single iteration of the Spectral Methods algorithm to find the principal eigenvector of the regularized weighted covariance matrix:
+    
+    .. math::
+        \begin{equation*}
+        M = A^H diag(T(y)) A + \lambda I,
+        \end{equation*}
+    
+    where :math:`A` is the forward matrix of the phase retrieval operator, :math:`T(\cdot)` is a preprocessing function for the measurements, and :math:`I` is the identity matrix of corresponding dimensions. Parameter :math:`\lambda` tunes the strength of regularization.
+
+    The iteration is given by
+
+    .. math::
+        \begin{equation*}
+        \begin{aligned}
+        x_{k+1} &= (M + \lambda I)x_k \\
```

Comment:
M already has the \lambda I, so you can remove the regularization term here (x_k+1 = M x_k)